### PR TITLE
[ADD] Manufacturing: By-Products

### PR DESCRIPTION
**Pull request by hojo-odoo**
This PR adds a new doc detailing how to add by-products to a product's BoM.

### CODE REVIEW

**Comment by hojo-odoo** [comment]
Hey @pabr-odoo,

This PR is ready for the first round of review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   *scrap wood* are created in addition to the Rocking Chair. By designating the scrap wood as a
+   by-product on the rocking chair's BoM, Odoo tracks the on-hand count of scrap wood, as well as
+   rocking chairs.
+
+Configuration
+=============
+
+To specify by-products on a product's |BoM|, the *By-Products* setting must be enabled. To do so,
+navigate to :menuselection:`Manufacturing app --> Configuration --> Settings`, and enable the
+:guilabel:`By-Products` checkbox, located under the :guilabel:`Operations` heading.
```
Comment (pabr-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:25]: Feel free to disregard if this isn't common practice, but I had to save the changes after enabling the checkbox before moving on to the next section. Would you mention this here?

**Review by pabr-odoo — APPROVED** [review]
Hey @hojo-odoo just gave this a look through and followed the work flow. Great addition! I added one comment that's more of a question on whether it's common practice to mention saving changes or if it's implicit, but everything looks great. Cool trick using variables to replace the abbreviations too, I learned something new.

**Comment by hojo-odoo** [comment]
Hey @ksc-odoo,

This PR is ready for final review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
+===========
+By-Products
+===========
+
+.. |MO| replace:: :abbr:`MO (manufacturing order)`
+.. |BoM| replace:: :abbr:`BoM (bill of materials)`
+.. |BoMs| replace:: :abbr:`BoMs (bills of materials)`
+
+When manufacturing certain products, it is common to be left with residual materials, in addition to
+the finished product. These materials are know as *by-products*. By specifying the by-products
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:10]:
```suggestion
the finished product. These materials are known as *by-products*. By specifying the by-products
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+created during manufacturing on a product's bill of materials (BoM), the quantity of by-products
+on-hand is tracked by Odoo.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:12]:
```suggestion
created during manufacturing on a product's bill of materials (BoM), the quantity of on-hand by-products
is tracked by Odoo.
```
SUGGESTION: slight reworking of the sentence to enhance the overall flow...

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+.. example::
+   Manufacturing a *rocking chair* requires ten pieces of wood. During production, five pieces of
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:15]:
```suggestion
   Manufacturing a *Rocking Chair* requires ten pieces of wood. During production, five pieces of
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+   *scrap wood* are created in addition to the Rocking Chair. By designating the scrap wood as a
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:16]:
```suggestion
   *scrap wood* are created, in addition to the Rocking Chair. By designating the scrap wood as a
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+   by-product on the rocking chair's BoM, Odoo tracks the on-hand count of scrap wood, as well as
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:17]:
```suggestion
   by-product on the Rocking Chair's BoM, Odoo tracks the on-hand count of scrap wood, as well as
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+   rocking chairs.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:18]:
```suggestion
   the number of Rocking Chairs.
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+Configuration
+=============
+
+To specify by-products on a product's |BoM|, the *By-Products* setting must be enabled. To do so,
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:23]:
```suggestion
To specify by-products on a product's |BoM|, the *By-Products* setting **must** be enabled. To do so,
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+navigate to :menuselection:`Manufacturing app --> Configuration --> Settings`, and enable the
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:24]:
```suggestion
navigate to :menuselection:`Manufacturing app --> Configuration --> Settings`, and tick the
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+Operation` field.
+
+.. image:: byproducts/byproducts-tab.png
+   :align: center
+   :alt: The By-Products tab on a BoM, configured with a "Scrap Wood" by-product.
+
+Manufacture by-product
+======================
+
+When an |MO| is completed and marked as *Done*, Odoo registers the quantity of by-products created
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:57]:
```suggestion
When an |MO| is completed, and marked as *Done*, Odoo registers the quantity of by-products created
```
OPTIONAL: added comma to enhance sentence structure/flow

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+during the manufacturing process. To create a new |MO|, navigate to :menuselection:`Manufacturing
+app --> Operations --> Manufacturing Orders`, and click :guilabel:`New`.
+
+In the :guilabel:`Bill of Material` field, select a |BoM| on which by-products have been specified.
+After doing so, the :guilabel:`Product` field auto-populates with the corresponding product. Click
+:guilabel:`Confirm` to confirm the |MO|.
+
+When manufacturing is completed, click the :guilabel:`Produce All` button at the top of the |MO|.
+After doing so, inventory counts update to reflect the quantity of by-product(s) produced as well as
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:66]:
```suggestion
After doing so, inventory counts update to reflect the quantity of by-product(s) produced, as well as
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+the quantity of the product.
+
+Click the :guilabel:`Product Moves` smart button at the top of the |MO| page to see the movements of
+components and products. Each by-product is listed on the page, with the :guilabel:`From` field
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:70]:
```suggestion
components and products. Each by-product is listed on the resulting :guilabel:`Inventory Moves` page, with the :guilabel:`From` column
```

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+displaying the virtual production location, and the :guilabel:`To` field displaying the location
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:71]:
```suggestion
displaying the virtual production location, and the :guilabel:`To` column displaying the location
```

**Review by ksc-odoo — APPROVED** [review]
hey @hojo-odoo -- just finished my Final Review. Nice work. Just a handful of minor suggestions/comments that require your attention. Approving now. Once you implement the necessary changes, feel free to tag this for Tech Review. Thanks! 👍

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

This PR is ready for technical review.

Thanks,
John

File: content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+When manufacturing certain products, it is common to be left with residual materials, in addition to
+the finished product. These materials are known as *by-products*. By specifying the by-products
+created during manufacturing on a product's bill of materials (BoM), the quantity of by-products
+on-hand is tracked by Odoo.
+
+.. example::
+   Manufacturing a *rocking chair* requires ten pieces of wood. During production, five pieces of
+   *scrap wood* are created, in addition to the rocking chair. By designating the scrap wood as a
+   by-product on the rocking chair's BoM, Odoo tracks the on-hand count of scrap wood, as well as
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/manufacturing/workflows/byproducts.rst:17]: Quick abbr add :)
```suggestion
   by-product on the rocking chair's |BoM|, Odoo tracks the on-hand count of scrap wood, as well as
```

**Review by samueljlieber — APPROVED** [review]
Hi @hojo-odoo! Awesome job on this new Manufacturing doc on By-products! I really like the example you used to explain what by-products are 👏 Approving with one small fix, thank you for your hard work!
.....
@robodoo delegate=hojo-odoo

**Comment by hojo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**